File: src/main/java/no/novari/flyt/authorization/client/sourceapplications/model/AcosSourceApplication.java
package no.novari.flyt.authorization.client.sourceapplications.model;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.stereotype.Component;

@Component
public class AcosSourceApplication extends BaseSourceApplication {
    public AcosSourceApplication(
            @Value("${fint.flyt.acos.sso.client-id:#{null}}") String clientId,
            @Value("${fint.flyt.acos.available:true}") boolean available
    ) {
        super(1L, "Acos Interact", clientId, available);
    }
}

File: src/main/kotlin/no/novari/flyt/authorization/client/sourceapplications/model/AcosSourceApplication.kt
package no.novari.flyt.authorization.client.sourceapplications.model

import org.springframework.beans.factory.annotation.Value
import org.springframework.stereotype.Component

@Component
class AcosSourceApplication(
    @Value("\${fint.flyt.acos.sso.client-id:#{null}}")
    clientId: String?,
    @Value("\${fint.flyt.acos.available:false}")
    available: Boolean,
) : BaseSourceApplication(
        id = 1L,
        displayName = "Acos Interact",
        clientId = clientId,
        available = available,
    )
